wgpu-core/src/command/query.rs

```diff
@@ -404,38 +404,41 @@ impl Global {
             .check_usage(wgt::BufferUsages::QUERY_RESOLVE)
             .map_err(ResolveError::MissingBufferUsage)?;
 
-        let end_query = start_query + query_count;
-        if end_query > query_set.desc.count {
+        let end_query = u64::from(start_query)
+            .checked_add(u64::from(query_count))
+            .unwrap();
```

Comment:
This should only ever need 33 bits

Reply:
Right...which is 1 more bit than the `u32`s here can hold, so I opted for widening to `u64`, preserving the potentially overflowed value for the diagnostic, and then narrowing back down for actual usage later. Is...this a request to change something?

Reply:
Added an `expect` message here. I'd be happy to downgrade this to a more simple arithmetic expression with a comment, though.